designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt

```diff
@@ -33,22 +54,165 @@ import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+                                    )
+                                }
+                            } else {
+                                Text(
+                                    text = noResultsFoundString,
+                                    textAlign = TextAlign.Center,
+                                    modifier = Modifier
+                                        .fillMaxWidth()
+                                        .padding(24.dp),
+                                )
+                            }
+                        }
+                    },
+                    onDismiss = {
+                        showDropdown = false
+                    },
+                    searchQuery = if (showSearchBar) {
+                        searchQuery
+                    } else {
+                        null
+                    },
+                    onSearch = { searchQuery = it },
+                    onSearchQueryChanged = { searchQuery = it },
+                )
+            }
+        }
+    } else {
+        ExposedDropdownMenuBox(
+            expanded = showDropdown,
+            onExpandedChange = { },
+            modifier = Modifier
+                .background(
+                    color = SurfaceColor.SurfaceBright,
+                    shape = RoundedCornerShape(Spacing8),
+                ),
+        ) {
+            inputField(modifier.menuAnchor())
+
+            MaterialTheme(
+                shapes = Shapes(extraSmall = RoundedCornerShape(Spacing8)),
+            ) {
+                ExposedDropdownMenu(
+                    expanded = showDropdown,
+                    onDismissRequest = { showDropdown = false },
+                    modifier = Modifier.background(
+                        color = SurfaceColor.SurfaceBright,
+                        shape = RoundedCornerShape(Spacing8),
+                    ).testTag("INPUT_DROPDOWN_MENU"),
+                ) {
+                    Column(modifier = Modifier.padding(horizontal = Spacing8)) {
+                        dropdownItems.forEachIndexed { index, item ->
+                            DropdownItem(
+                                modifier = Modifier.testTag("INPUT_DROPDOWN_MENU_ITEM_$index"),
+                                item = item,
+                                selected = selectedItem == item,
+                                contentPadding = PaddingValues(
+                                    horizontal = Spacing8,
+                                    vertical = Spacing16,
+                                ),
+                                onItemClick = {
+                                    onItemSelected(item)
+                                    showDropdown = false
+                                },
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun InputField(
```

Comment:
What do you think of renaming this class to something like DropdownInputField? we already have a file that's called InputField that contains the BasicTextField we use in most inputs